Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -1264,10 +1280,13 @@ extension DiffContainerViewController: DiffToolbarViewDelegate {
               let username = toModel?.user else {
             return
         }
-        
+
         fakeProgressController.start()
-        WKWatchlistService().rollback(title: title, project: wkProject, username: username) { [weak self] result in
-            DispatchQueue.main.async {
+        DispatchQueue.main.async {
```

Comment:
I'm getting a threading error after you moved this dispatch queue call. Can you move it to only directly surround the `self?.completeRollbackOrUndo` call (line 1290)?

The rollback call to the service sends it to a different thread, so even if you have it above on line 1285 it won't remain on the main thread in the service completion handler.